src/main/java/ralf2oo2/freecam/mixin/EntityMixin.java

```diff
@@ -55,17 +51,17 @@ private void freecam_cameraMovementHandler(float yaw, float pitch, CallbackInfo
 
         // Update freecam speed
         if(Freecam.freecamController.updateSpeed){
-            Freecam.config.speed += yaw / 10;
-            Freecam.config.speed = (float)(new BigDecimal(Freecam.config.speed).setScale(1, RoundingMode.HALF_UP).doubleValue());
-            if(Freecam.config.speed < 0){
-                Freecam.config.speed = (float)0;
+            FreecamConfig.config.speed += yaw / 10;
+            FreecamConfig.config.speed = (float)(new BigDecimal(FreecamConfig.config.speed).setScale(1, RoundingMode.HALF_UP).doubleValue());
+            if(FreecamConfig.config.speed < 0){
+                FreecamConfig.config.speed = (float)0;
             }
-            if(Freecam.config.speed > 1000){
-                Freecam.config.speed = (float)1000;
+            if(FreecamConfig.config.speed > 1000){
+                FreecamConfig.config.speed = (float)1000;
             }
-            JsonObject jsonObject = new JsonObject();
-            jsonObject.put("speed", new JsonPrimitive(Freecam.config.speed));
-            GCAPI.reloadConfig(Identifier.of("freecam:config"), jsonObject);
+            GlassYamlFile modConfigFile = new GlassYamlFile();
+            modConfigFile.set ("speed", FreecamConfig.config.speed);
+            GCAPI.reloadConfig(Identifier.of("freecam:config").toString(), modConfigFile);
```

Comment:
This is the only piece of code that I am uncertain if it works or not.
If you can let me know how to test this, that would be great :)

Reply:
It saves the speed value when changed in game. I just happened to also be working on updating freecam to GCAPI3 but got stuck at migrating from 2.0, I'll merge this and check if it works. Thx for the help

Reply:
Yep no problem. I mainly wanted to update it because I am about to release my local server mod, which needed FreeCam to not crash the server when in the server mods folder :slightly_smiling_face: